pkg/controller/controller.go

```diff
@@ -343,13 +355,27 @@ func (c *Controller) scaleNodeGroup(nodegroup string, nodeGroup *NodeGroupState)
 		// if ScaleUpThresholdPercent is our "max target" or "slack capacity"
 		// we want to add enough nodes such that the maxPercentage cluster util
 		// drops back below ScaleUpThresholdPercent
-		nodesDelta, err = calcScaleUpDelta(untaintedNodes, cpuPercent, memPercent, cpuRequest, memRequest, nodeGroup)
+		nodesDelta, err = calcScaleUpDelta(
+			untaintedNodes,
+			cpuPercent,
+			memPercent,
+			*podRequests.Total.GetCPUQuantity(),
+			*podRequests.Total.GetMemoryQuantity(),
+			nodeGroup)
 		if err != nil {
 			log.Errorf("Failed to calculate node delta: %v", err)
 			return nodesDelta, err
 		}
 	}
 
+	if nodeGroup.Opts.ScaleOnStarve &&
+		((!podRequests.LargestPendingCPU.IsEmpty() && podRequests.LargestPendingCPU.MilliCPU > nodeCapacity.LargestAvailableCPU.MilliCPU) ||
```

Comment:
nit: this condition is getting a bit long. Perhaps breaking the second part into a dedicated function
```go
&& ((!podRequests.LargestPendingCPU.IsEmpty() && podRequests.LargestPendingCPU.MilliCPU > nodeCapacity.LargestAvailableCPU.MilliCPU) ||
			(!podRequests.LargestPendingMemory.IsEmpty() && podRequests.LargestPendingMemory.Memory > nodeCapacity.LargestAvailableMemory.Memory))
```

Reply:
Probably a bit more than a nit really. I 100% agree and will do this. Only reason I hadn't already is because the parent function is already so large.